privacyidea/api/lib/prepolicy.py

```diff
@@ -693,6 +693,8 @@ def check_max_token_user(request=None, action=None):
     ERROR_ACTIVE_TYPE = "The number of active tokens of type {0!s} for this user is limited!"
     params = request.all_data
     user_object = get_user_from_param(params)
+    if not user_object.login and params.get("serial", None):
+        user_object = get_token_owner(params["serial"])
     if user_object.login:
```

Comment:
In case a serial is given but there is no user assigned to the token, this fails since `user_object` will be `None`. We could add `if user_object and user_object.login:` to mitigate this.

Reply:
If the token is not assigned, the max_token policy should not block anything...?

I added a test for this, but in my opinion, it works as it should

Reply:
The problem is that this pre-policy function will always be executed regardless of an existing max_token policy (the policy will be queried later).
So in case we only have a serial and the token does not have a user assigned, the `get_token_owner(serial)` function returns 'None' and the check breaks.

Reply:
if the token with serial does not exist yet (as in /token/init), get_token_owner(serial) does throw an exception instead of returning None, since the token does not exists yet.